### PR TITLE
Implement of Get-AzureRmVMAccessExtension linux vm support.

### PR DESCRIPTION
**Pull request by takekazuomi**
## Description
Fix 
https://github.com/Azure/azure-powershell/issues/3450

Linux VM's VMAccessExtension  use diffrent pulisher and type name. Add string constant and check it.
```
    "publisher": "Microsoft.OSTCExtensions",
    "type": "VMAccessForLinux",
```

**NOTE**
Set-AzureRmVMAccessExtension**  still have problem about linux vm.
This code did not fix it take care.

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@takekazuomi__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, AZPRBOT;

**Comment by cormacpayne** [comment]
@eshaparmar Hey Esha, would you mind reviewing this PR when you get the chance?

**Comment by eshaparmar** [comment]
@cormacpayne we dont own these cmdlets so have less context. if you still want me to then i can look at it.

**Comment by shahabhijeet** [comment]
@takekazuomi waiting for the Extensions team to approve the PR.

File: src/ResourceManager/Compute/Commands.Compute/Extension/VMAccess/VirtualMachineAccessExtensionContext.cs [hunk 1]
```diff
@@ -23,6 +23,9 @@ public class VirtualMachineAccessExtensionContext : PSVirtualMachineExtension
         public const string ExtensionDefaultPublisher = "Microsoft.Compute";
         public const string ExtensionDefaultName = "VMAccessAgent";
         public const string ExtensionDefaultVersion = "2.0";
+        public const string ExtensionLinuxPublisher = "Microsoft.OSTCExtensions";
+        public const string ExtensionLinuxName = "VMAccessForLinux";
+        public const string ExtensionLinuxVersion = "2.0";
```
Comment (boumenot) [inline src/ResourceManager/Compute/Commands.Compute/Extension/VMAccess/VirtualMachineAccessExtensionContext.cs]: Why is this 2.0?  The latest version of VMAccessForLinux is 1.4.6.0.

How was this tested?

Comment (hyonholee) [inline src/ResourceManager/Compute/Commands.Compute/Extension/VMAccess/VirtualMachineAccessExtensionContext.cs]: +1  why is this "2.0"?  Shouldn't it be "1.4" ?

Comment (takekazuomi) [inline src/ResourceManager/Compute/Commands.Compute/Extension/VMAccess/VirtualMachineAccessExtensionContext.cs]: This is my mistake.
1.4 is correct.

**Comment by boumenot** [comment]
I don't know this code base well enough, but how does it enable auto upgrade of minor version for extensions? If VMAccessForLinux 1.5 is released, do I need to rev this code to pick up that version?  This needs to be addressed, but perhaps outside of this commit.

**Comment by huangpf** [comment]
Why it's only touching the Get- related parts. How about the Set- part? 

Are all parameters the same for enabling both Windows and Linux VMAccess extensions?

**Comment by takekazuomi** [comment]
About SET- operation in following document,
https://github.com/Azure/azure-linux-extensions/tree/master/VMAccess#22-using-azure-powershell

It is enough for Set-AzureVMExtension. Dedicated cmdlets that is Set-AzureRmVMAccessExtension is  unnecessary. 
Don't you need Get-AzureRmVMAccessExtension too ?

**Comment by markcowl** [comment]
@takekazuomi @huangpf Can we get to a consensus about this pull request?  Will have to close by EOW if there is no further activity

**Comment by cormacpayne** [comment]
Closing due to lack of activity for the last couple of weeks.

@takekazuomi @huangpf please feel free to reopen this PR at any point

**Pull request closed without merging**